plutus-core/plutus-core/src/PlutusCore/Builtin/Runtime.hs

```diff
@@ -59,11 +68,32 @@ instance NFData (RuntimeScheme val args res) where
 -- All the three are in sync in terms of partial instantiatedness due to 'TypeScheme' being a
 -- GADT and 'FoldArgs' and 'FoldArgsEx' operating on the index of that GADT.
 data BuiltinRuntime val =
-    forall args res. BuiltinRuntime
-        (RuntimeScheme val args res)
-        ~(FoldArgs args res)  -- Must be lazy, because we don't want to compute the denotation when
-                              -- it's fully saturated before figuring out what it's going to cost.
-        (FoldArgsEx args)
+    forall n. BuiltinRuntime
+        (RuntimeScheme n)
+        ~(ToDenotationType val n)  -- Must be lazy, because we don't want to compute the denotation
+                                   -- when it's fully saturated before figuring out what it's going
+                                   -- to cost.
+        (ToCostingType n)
+
+data BuiltinRuntimeOptions n val cost =
```

Comment:
can we make this a record type pleas so we can remember which `ToDenotationType` is which!

Reply:
Also worth a comment, in particular I forgot that `cost` here is the cost *model* and had to go looking at the use sites to remember.